tuf/api/exceptions.py

```diff
@@ -0,0 +1,68 @@
+# Copyright New York University and the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""
+Define TUF exceptions used inside the new modern implementation.
+The names chosen for TUF Exception classes should end in 'Error' except where
+there is a good reason not to, and provide that reason in those cases.
+"""
+
+
+#### General errors ####
+
+# Reexport securesystemslib exceptions so that users of code that throws them
+# doesn't have to import securesystemslib to handle them.
+# pylint: disable=unused-import
+from securesystemslib.exceptions import UnsupportedAlgorithmError
```

Comment:
You know what? We can get rid of this without affecting anyones quality of life:

The only real use of UnsupportedAlgorithmError is in `Targetfile.from_data()` and in there it happens if an _argument_ contains a bad algorithm name. I think we can just raise a ValueError there instead without losing anything. Then we would have no more securesystemslib errors in the public Metadata API AFAICT.